grid.v3/client.go

```diff
@@ -54,6 +50,122 @@ func (c *Client) Close() error {
 	return err
 }
 
+type PeerChangeEvent struct {
+	peer        string
+	err         error
+	stateChange int
+}
+
+func (p *PeerChangeEvent) Peer() string {
+	return p.peer
+}
+
+func (p *PeerChangeEvent) Discovered() bool {
+	return p.stateChange == PeerDiscovered
+}
+
+func (p *PeerChangeEvent) Lost() bool {
+	return p.stateChange == PeerLost
+}
+func (p *PeerChangeEvent) Err() error {
+	return p.err
+}
+
+const (
+	WatchErr       = -1
+	PeerDiscovered = 1
+	PeerLost       = 2
+)
+
+/*
+
+	client := grid.Client(...)
+	watch, currentpeers := client.PeersWatch(c)
+	for _, peer := range currentpeers {
+			//do work or request metadata from peer
+	}
+	for peer := range watch.C {
+			if peer.Err() != nil {
+				//deal with a peer that went down.. reschedule actors..
+			}
+			if peer.Lost() {
+				//deal with a peer that went down.. reschedule actors..
+			}
+			if peer.Discovered() {
+				//Request metadata from the peer.
+				//deal with a new peer, maybe reschedule existing actors on it it..
+			}
+	}
+*/
+func (c *Client) PeersWatch(ctx context.Context) ([]string, <-chan *PeerChangeEvent, error) {
+	peers, err := c.PeersC(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	currentPeers := map[string]struct{}{}
+	for _, p := range peers {
+		currentPeers[p] = struct{}{}
+	}
+
+	watchchan := make(chan *PeerChangeEvent)
+	go func() {
+		//TODO consider switching this to using an etcd watch? But that maybe overkill and more complex?
+		ticker := time.NewTicker(2 * time.Second)
+		defer ticker.Stop()
+
+		oldPeers := currentPeers
+		for {
+			select {
+			case <-ticker.C:
+				peers, err := c.PeersC(ctx)
+				if err != nil {
+					close(watchchan)
+					watchchan <- &PeerChangeEvent{"", err, WatchErr}
```

Comment:
We probabbly need to make these:

```
select {
case watchchan <- &PeerChangeEvent{"", err, WatchErr}:
case <-ctx.Done():
    return
}
```

Also looks like we can make the `close(watchchan)` a single defer?